spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -799,6 +799,9 @@ def write_data_spec(
 
         self._weight_scales[placement] = weight_scales
 
+    def clear_connection_cache(self):
+        self._retrieved_blocks = dict()
```

Comment:
No, this isn't a blocker, more just a thought (though if not here, it probably won't happen ;).  You are right that structural plasticity makes this more complicated, though it can equally detect this and clear if structural plasticity is in place.  Once this is merged, a ticket can be created (once the other change is done ;).